tests/e2e/specs/setup-mc/step-1-accounts.test.js

```diff
@@ -218,6 +211,152 @@ test.describe( 'Set up accounts', () => {
 
 			expect( page.url() ).toMatch( baseURL + 'google_auth' );
 		} );
+
+		test( 'should create merchant center and ads account if does not exist for the user', async () => {
+			await setUpAccountsPage.mockJetpackConnected();
+			await setUpAccountsPage.mockGoogleConnected();
+			await setUpAccountsPage.mockAdsAccountDisconnected();
+			await setUpAccountsPage.mockMCNotConnected();
+
+			await setUpAccountsPage.fulfillAdsAccounts(
+				[
+					[],
+					[
+						{
+							id: 78787878,
+							name: 'gla',
+						},
+					],
+				],
+				200,
+				[ 'GET' ],
+				true
+			);
+
+			await setUpAccountsPage.fulfillMCAccounts(
+				[
+					[],
+					[
+						{
+							id: 5432178,
+							name: null,
+							subaccount: null,
+							domain: null,
+						},
+					],
+				],
+				200,
+				'GET',
+				true
+			);
+
+			await setUpAccountsPage.fulfillAdsConnection(
+				[
+					{
+						id: 0,
+						currency: 'USD',
+						status: 'disconnected',
+						symbol: '$',
+					},
+					{
+						id: 78787878,
+						currency: 'USD',
+						status: 'incomplete',
+						step: 'account_access',
+						sub_account: true,
+						symbol: '$',
+					},
+				],
+				200,
+				'GET',
+				true
+			);
+
+			await setUpAccountsPage.fulfillMCConnection(
+				[
+					{
+						id: 0,
+						name: null,
+						subaccount: null,
+						domain: null,
+					},
+					{
+						id: 5432178,
+						name: null,
+						subaccount: null,
+						domain: null,
+						status: 'incomplete',
+						step: 'claim',
+					},
+				],
+				200,
+				'GET',
+				true
+			);
+
+			await setUpAccountsPage.goto();
+			const googleAccountCard = setUpAccountsPage.getGoogleAccountCard();
+
+			await expect(
+				googleAccountCard.getByText(
+					/You don’t have Merchant Center nor Google Ads accounts, so we’re creating them for you./,
+					{
+						exact: true,
+					}
+				)
+			).toBeVisible();
+		} );
+
+		test.describe( 'After connecting Google account', () => {
+			test.beforeEach( async () => {
+				await setUpAccountsPage.mockJetpackConnected();
+				await setUpAccountsPage.mockGoogleConnected();
+				await setUpAccountsPage.mockMCConnected();
+				await setUpAccountsPage.mockAdsAccountConnected();
+
+				await setUpAccountsPage.goto();
+			} );
+
+			test( 'should see the merchant center id and ads account id if connected', async () => {
+				const googleAccountCard =
+					setUpAccountsPage.getGoogleAccountCard();
+				await expect(
+					googleAccountCard.getByText( 'Merchant Center ID: 1234', {
+						exact: true,
+					} )
+				).toBeVisible();
+
+				await expect(
+					googleAccountCard.getByText( 'Google Ads ID: 12345', {
+						exact: true,
+					} )
+				).toBeVisible();
+			} );
+
+			test( 'should see the connected label', async () => {
```

Comment:
The UI errors mentioned in https://github.com/woocommerce/google-listings-and-ads/pull/2618#pullrequestreview-2391575474 still happen.